.github/workflows/ci.yaml

```diff
@@ -30,16 +30,16 @@ jobs:
       - name: Check Title Format
         uses: gsactions/commit-message-checker@v2
         with:
-          pattern: '\[(DOCS|FEAT|FIX|STYLE|DEVOPS|REFACTOR|PERF|TEST|CHORE)\] .+$'
+          pattern: '\[(FEAT|FIX|DOCS|STYLE|REFACTOR|PERF|TEST|CHORE|DEVOPS|REVERT)\] .+$'
           error: 'Your PR title must contain a commit type like "[FIX]".'
           excludeDescription: 'false'  # Check the entire commit message
           excludeTitle: 'false'  # Do not exclude the title
           checkAllCommitMessages: 'false'  # Check only the PR title, assuming it's the first commit
-        if: github.event_name == 'pull_request' && ${{ !github.event.pull_request.merged }} && ${{ !cancelled() }}
+        if: ${{ github.event_name == 'pull_request' && !cancelled() }}
```

Comment:
### Fix YAML syntax

The whole first if condition was not working correctly because all `!` have to be escaped.
From my testing it also looks like logical operators cannot combine multiple `${{ }}` expressions - they have to be inside the `${{ }}`.

Putting all the expressions into one `${{ }}` block should fix the ignored check for `github.event_name == 'pull_request'` this time.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idif